feubot.py

```diff
@@ -4,38 +4,730 @@
+TOKEN = os.getenv('DISCORD_TOKEN')
+
+bot = commands.Bot(command_prefix="!")
+
+bot = Bot("!")
+
+@bot.event
+async def on_ready():
+    print(f'{bot.user} has connected to Discord!')
+
 
-import helpful, memes, reactions
 from feubotFormatter import FeubotFormatter
 
+def setupBot(bot):
+
+    #Reload this as part of reload due to use of other.developerCheck
+    @bot.command(pass_context=True, hidden = True, aliases = ['exec'])
+    @other.developerCheck
+    async def debug(ctx, *, arg):
+        # https://stackoverflow.com/questions/3906232/python-get-the-print-output-in-an-exec-statement
+        from io import StringIO
+        import sys
+        old_stdout = sys.stdout
+        redirected_output = sys.stdout = StringIO()
+        bot = ctx.bot
+        try:
+            exec(arg)
+        except SystemExit:
+            await bot.say("I tried to quit().")
+        finally:
+            sys.stdout = old_stdout
+        output = redirected_output.getvalue()
+        output = "No output." if not output else output
+        await bot.say(output)
+
+
 if __name__ == "__main__":
     if "--debug" in argv:
-        bot = commands.Bot(command_prefix=['##', 'feubeta '], description='this is feubot beta.', formatter = FeubotFormatter()) 
+        bot = commands.Bot(command_prefix=['##', 'feubeta '], description='this is feubot beta.', formatter = FeubotFormatter())
     else:
-        bot = commands.Bot(command_prefix=['!', '>>', 'feubot '], description='this is feubot.', formatter = FeubotFormatter())
+        bot = commands.Bot(command_prefix=['!', '>>', 'feubot '], description="Hi I'm feubot. I'm case sensitive.", formatter = FeubotFormatter())
+
+    def trunc_to(ln, s):
+        if len(s) >= ln: return s
+        else: return s[:ln-3] + "..."
+
+    def create_embed(posts, threads, term):
+        feu_search_base = "http://feuniverse.us/search?q=%s"
+        feu_post_base = "http://feuniverse.us/t/{}/{}"
+
+        result = discord.Embed(
+                title="Search results",
+                url=feu_search_base % term,
+                description="Found %d results" % len(posts),
+                color=0xde272c)
+        for i,post in enumerate(posts[:5]):
+            result.add_field(
+                    name='Post in "%s" by %s' % (threads[i]["title"], post["name"]),
+                    value="[%s](%s)" %
+                        (trunc_to(50, post["blurb"]),
+                         feu_post_base.format(post["topic_id"], post["post_number"])),
+                    inline=False)
+        if len(posts) > 5:
+            result.set_footer(text="Truncated %d result(s)." % (len(posts)-5))
+        return result
+
+
 
     @bot.event
     async def on_ready():
         print('Logged in as')
         print(bot.user.name)
         print(bot.user.id)
         print('------')
-        await bot.change_presence(game=discord.Game(name="Reading the doc!"))
+        #await bot.change_presence(status=discord.Status.idle, activity=game)
```

Comment:
leaving these in just generates more clutter in an already really messy codebase. just remove them